### PR TITLE
add docs for kustomize schema generator

### PR DESCRIPTION
**Pull request by y-ykcir**

### CODE REVIEW

**Comment by y-ykcir** [comment]
/hold
Need to add the correct path to the docs.

**Comment by y-ykcir** [comment]
@zmberg PTAL

File: docs/cli-tool/kustomize-schema-generator.md [hunk 1]
```diff
@@ -0,0 +1,170 @@
+---
+title: Kustomize Schema Generator
+---
+
+
+[kruise-api](https://github.com/openkruise/kruise-api) provides tools for openkruise to generate `kustomize schema` files that support openkruise CRD, and provides the current available `OpenAPI schema`.
+
+## Schema Generator
+
+When using `kustomize` to manage applic ations, in order to use `strategic merge patches (SMPs)` to process environment variables or any array type fields in resource definitions when using openkruise resources, the `OpenAPI schema` of openkruise CRD needs to contain the specified `patch strategy`. `Schema Generator` supports the function of quickly generating `kustomize schema` files based on the resource definition of openkruise. The specific way of generating `kustomize schema` files can be viewed in [README.md](https://github.com/openkruise/kruise-api/blob/master/cmd/gen-schema/README.md).
```
Comment (furykerry) [inline docs/cli-tool/kustomize-schema-generator.md:10]: applic ations -> applications

File: docs/cli-tool/kustomize-schema-generator.md [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+```yaml
+openapi:
+  path: <kruise-api-local-path>/schema/openkruise_all_k8s_kustomize_schema.json
+```
+
+### Example
+
+Take `sidecarset` as an example to introduce how to use `kustomize OpenAPI schema` to support the merge strategy of openkruise CRD in kustomize.
+
+A complete definition of `sidecarset` is as follows:
```
Comment (furykerry) [inline docs/cli-tool/kustomize-schema-generator.md:43]: plz illustrate how to delete an entry in the array.

File: i18n/zh/docusaurus-plugin-content-docs/current/cli-tool/kustomize-schema-generator.md [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+```yaml
+openapi:
+  path: https://raw.githubusercontent.com/openkruise/kruise-api/master/schema/openkruise_all_k8s_kustomize_schema.json
+```
+
+使用指定版本
+
+```yaml
+openapi:
+  path: https://raw.githubusercontent.com/openkruise/kruise-api/raw/<tag>/schema/argo_all_k8s_kustomize_schema.json
```
Comment (veophi) [inline i18n/zh/docusaurus-plugin-content-docs/current/cli-tool/kustomize-schema-generator.md:30]: > argo_all_k8s_kustomize_schema.json

File: docs/cli-tool/kustomize-schema-generator.md [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+```yaml
+openapi:
+  path: https://raw.githubusercontent.com/openkruise/kruise-api/master/schema/openkruise_all_k8s_kustomize_schema.json
+```
+
+Use the specified version
+
+```yaml
+openapi:
+  path: https://raw.githubusercontent.com/openkruise/kruise-api/raw/<tag>/schema/argo_all_k8s_kustomize_schema.json
```
[the remark this answers is not in the document]
Comment (y-ykcir) [reply docs/cli-tool/kustomize-schema-generator.md]: Fixed

File: docs/cli-tool/kustomize-schema-generator.md (hunk 2 above)
Comment (y-ykcir) [reply docs/cli-tool/kustomize-schema-generator.md:43]: Has illustrated

File: i18n/zh/docusaurus-plugin-content-docs/current/cli-tool/kustomize-schema-generator.md (hunk 3 above)
Comment (y-ykcir) [reply i18n/zh/docusaurus-plugin-content-docs/current/cli-tool/kustomize-schema-generator.md]: Fixed

**Review by furykerry — COMMENTED** [review]
/lgtm

**Pull request merged**